src/scripts.js

```diff
@@ -46,7 +65,94 @@ function initializeData(data) {
   let customerInfo = data[0];
   let roomInfo = data[1];
   let bookingInfo = data[2];
-  let customer = new Customer(customerInfo, bookingInfo, roomInfo);
+  separatedData = [customerInfo, roomInfo, bookingInfo];
+
+  customer = new Customer(customerInfo, bookingInfo, roomInfo);
+
   domUpdates.populateUserInfo(customer);
   domUpdates.populateUpcomingStays(customer);
+  domUpdates.populateRoomTypeDropDwn(roomInfo);
+};
+
+function createRooms(roomsToCreate) {
+  let instantiatedRooms = roomsToCreate.map((room) => {
+    let createdRoom = new Room(room);
+    return createdRoom;
+  });
+
+    domUpdates.populateFilteredRooms(instantiatedRooms, gridContainer, roomDisplayHeading, upcomingStaysBtn);
+    roomsOnDashboard = instantiatedRooms;
+};
+
+function filterAvailableRooms() {
+if (!checkinDate.value || !checkoutDate.value) {
+    domUpdates.show(dateError);
+    return
+  };
+
+  let allRoomInfo = separatedData[1];
+  let allBookingInfo = separatedData[2];
+  let roomNumbersAdded = [];
+
+  let availableRooms = allRoomInfo.rooms.reduce((arr, room) => {
+    allBookingInfo.bookings.forEach((booking) => {
+      let parsedBookingDate = booking.date.replaceAll('/', '');
+      let parsedCheckinDate = checkinDate.value.replaceAll('-', '');
+      roomNumbersAdded;
+      if (!roomNumbersAdded.includes(room.number) && booking.roomNumber === room.number && parsedCheckinDate !== parsedBookingDate) {
+        arr.push(room);
+        roomNumbersAdded.push(room.number);
+      };
+    });
+    return arr;
+  }, []);
+
+  if (roomType.value !== 'All') {
+    let filteredByType = availableRooms.filter((room) => room.roomType === roomType.value);
+    if (!filteredByType.length) {
+      domUpdates.show(noResultsMsg);
+    }
+    createRooms(filteredByType);
+
+  } else if (roomType.value === 'All') {
+    if (!availableRooms.length) {
+      domUpdates.show(noResultsMsg);
+    }
+    createRooms(availableRooms);
+  };
+};
+
+function findRoom(event) {
```

Comment:
I really like what you have here - it looks like you have a function to display info about the room - I am missing this in mine.